text/0000-clarify-api-namespace.md

```diff
@@ -0,0 +1,59 @@
+- Start Date: 2015-08-10
+- RFC PR: (leave this empty)
+- Refract Issue: (leave this empty)
+
+# Summary
+Clarify API description namespace and Iron out issues which rose during API Blueprint serialization to API description namespace.
+
+# Motivation
+The purpose of this document is to clear inconsistencies in the format and make it easier to build tools around it.
+
+# Detailed design
+
+## HREF Variables
+We need to clarify about the value of the `content` in the `hrefVariables` element.
+
+## Data Structure
+We don't need a `Data Structure` because an element which is an `object`, `extend` or `array` can be considered as a data structure element.
```

Comment:
> cause an element which is an `object`, `extend` or `array`

Note it can be any time (even `string` etc).

You are correct that as it is defined here it does not make much sense. This element was supposed to represent "Attributes" section in API Blueprint. Correctly it should be:

```apib
# Data Structure (Element)
+ element: dataStructure (string, fixed)
+ content (Data Structure Element) 
```

so, for example: 

```json
["dataStructure", {}, {}, [
  "string", {}, {}, "Hello World!"
  ]]
```

The reason to have a separate `Data Structure` element instead of just directly a *Data Structure Namespace* Data Structure Element` was to explicitly state "this is data structure defined to represent the (parent) element" versus just having an arbitrary `object` or `string` lying there:

Consider:

```json
["resource", {}, {}, [
  ["dataStructure", {}, {}, [
    "string", {}, {}, "Hello World!"
    ]]
  ]]
```

vs.

```json
["resource", {}, {}, [
  "string", {}, {}, "Hello World!"
  ]]
```

Thoughts?